app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -145,20 +145,25 @@ def validate_max_members_not_exceeded!(group)
   # rubocop:disable Metrics/AbcSize
   # rubocop:disable MethodLength
   def create_group_assignment_repo(selected_group: group, new_group_title: nil)
-    if !invitation.enabled?
-      flash[:error] = "Invitations for this assignment have been disabled."
+    result = invitation.redeem_for(
+      current_user,
+      selected_group,
+      new_group_title,
+      group_import_resiliency: group_import_resiliency_enabled?
+    )
+
+    case result.status
+    when :failed
+      GitHubClassroom.statsd.increment("group_exercise_invitation.fail")
```

Comment:
Small note here: We're now counting invitations not being enabled as a failure, whereas before we weren't. Not sure that case should count as a failure, what do you think?